### PR TITLE
Adds ectoscopic sniffer

### PR DESCRIPTION
**Pull request by Dejaku51**
## About The Pull Request

Original PR: https://github.com/tgstation/tgstation/pull/59494
This PR adds a new machine; the ectoscopic sniffer, to the robotics lab on each map.

The sniffer is small desktop machine used for detecting paranormal activity, it displays a green flash and a chime every time it is touched(as in attacked) by a ghost.

It has a single wire inside, when this wire is pulsed, it triggers the animation and chime, as if touched by a ghost, maybe some enterprising crewmember can use this to their advantage. When this wire is cut, the device won't trigger, no matter how many spooks or haunts touch it.

## Why It's Good For The Game
Ghosts can now inform roboticists that they are supposed to make borgs 

## Changelog
:cl:
add: Added a new machine to the robotics lab; the ectoscopic sniffer.
/:cl:

### CODE REVIEW

**Comment by DatBoiTim** [comment]
Mmm yes can't wait for ouija boards v2: Morse code

**Comment by Troontjelolo** [comment]
>Mmm yes can't wait for ouija boards v2: Morse code

.. -. ... . .-. - / -. .- -- . / .. ... / .- / .-.. .. -. --.

**Comment by DatBoiTim** [comment]
> > Mmm yes can't wait for ouija boards v2: Morse code
> 
> .. -. ... . .-. - / -. .- -- . / .. ... / .- / .-.. .. -. --.

You joke but you can build these things.

Left is Dot, Right is Dash, Middle is New Word. All you need to do, is get ghosts attention and tell them how to use it.

You can easily find translators for morse code on the internet, so this really won't require any special skill.

**Comment by Isy232** [comment]
> > > Mmm yes can't wait for ouija boards v2: Morse code
> > 
> > 
> > .. -. ... . .-. - / -. .- -- . / .. ... / .- / .-.. .. -. --.
> 
> You joke but you can build these things.
> 
> Left is Dot, Right is Dash, Middle is New Word. All you need to do, is get ghosts attention and tell them how to use it.
> 
> You can easily find translators for morse code on the internet, so this really won't require any special skill.

Ghosts could just feed false information for the fun of it.

Its also not really easy to use it for morse code, since its difficult to do dots and dashes with it. It would require a lot of time just to get a small message through.

Still, maybe a little iffy.

**Comment by DatBoiTim** [comment]
> Ghosts could just feed false information for the fun of it.

Not really any better.
Could lead to people getting round removed over a false ling accusation or such

Also, as mentioned, building multiple can make for a much better morse code setup.

**Comment by Dejaku51** [comment]
You know that there is cooldown for every ghost to avoid spamming?

**Comment by DatBoiTim** [comment]
> You know that there is cooldown for every ghost to avoid spamming?

Per thing, yeah

**Comment by St0rmC4st3r** [comment]
> Not really any better. Could lead to people getting round removed over a false ling accusation or such

What would you do as a policeman if some rando would come to you and say that the cook in the cafeteria is a murderer because ghosts told you about it?

**Comment by DatBoiTim** [comment]
> > Not really any better. Could lead to people getting round removed over a false ling accusation or such
> 
> What would you do as a policeman if some rando would come to you and say that the cook in the cafeteria is a murderer because ghosts told you about it?

A) This item is designed to have ghosts acknowledged in round, so far to the point it is meant to guide the decision of whether or not Robotics builds posibrains. IRL, manners of communicating with ghosts are sketchy at best.

B) It's happened before. HELL people have used the number of ghosts orbiting people as grounds for shit. This kind of shit is why we removed the ghost vision  goggles.

File: code/game/machinery/ecto_sniffer.dm
Comment (PowerfulBacon) [inline code/game/machinery/ecto_sniffer.dm]: Could this be replaced with user.ckey in ectoplasmic_residues or an assoc list instead to do a dictionary access?
```
ectoplasmic_residues[ckey] = TRUE
if(ectoplasmic_residues[ckey])
   ...
```

**Review by PowerfulBacon — DISMISSED** [review]

**Comment by MNarath1** [comment]
> > > Mmm yes can't wait for ouija boards v2: Morse code
> 
> > 
> 
> > .. -. ... . .-. - / -. .- -- . / .. ... / .- / .-.. .. -. --.
> 
> 
> 
> You joke but you can build these things.
> 
> 
> 
> Left is Dot, Right is Dash, Middle is New Word. All you need to do, is get ghosts attention and tell them how to use it.
> 
> 
> 
> You can easily find translators for morse code on the internet, so this really won't require any special skill.

To be fair there is other ways to do the same too.

**Comment by utilbird** [comment]
"Pulse twice if X killed you"

**Comment by MNarath1** [comment]
I would say the easiest way to address the problem of metagaming is to make it blink for like 30 seconds when pulsed once(so you actually can get the attention of whoever is there) and to put a cooldown on it so you can't possible use it to communicate information beyond we are here and we want to get back into the round

**Comment by DatBoiTim** [comment]
> and to put a cooldown on it so you can't possible use it to communicate

This will need to be significant enough so that you can't build multiple for an increased bit rate.

**Comment by Dejaku51** [comment]
PR completed, waiting for maintainers to merge it

**Comment by hotimep** [comment]
mfrw kill my murder obj but he posthumouslys rats me out with fucking morse code

**Comment by MNarath1** [comment]
> mfrw kill my murder obj but he posthumouslys rats me out with fucking morse code

there is an easy fix for that and i already did suggest it

**Comment by Dejaku51** [comment]
surely you can rat someone out with that 15 seconds cooldown

**Comment by Dejaku51** [comment]
I'm not gonna wait a month for someone to merge it. Feel free to create that PR yourself

**Comment by Dejaku51** [comment]
Fuck it, im just gonna leave it open. If any maintainer gonna review it please ping me on discord (Dejaku51#9677) so i can make changes

File: code/game/machinery/ecto_sniffer.dm
Comment (Archanial) [inline code/game/machinery/ecto_sniffer.dm:56]: ?

Comment (Archanial) [inline code/game/machinery/ecto_sniffer.dm]:
```suggestion
```
read it when it's actually ported

Comment (Archanial) [reply code/game/machinery/ecto_sniffer.dm]: Or add emissive overlays in much more annoying way ~Kube

File: code/game/machinery/ecto_sniffer.dm
Comment (Archanial) [inline code/game/machinery/ecto_sniffer.dm]:
```suggestion
	if(!on || !sensor_enabled || !is_operational())
```
micro op really

**Review by Archanial — APPROVED** [review]

**Pull request merged**
